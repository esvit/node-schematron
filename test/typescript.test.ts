import { sync } from 'slimdom-sax-parser';

import { Schema } from '../index';

describe('typescript', () => {
	let schema = null;

	it('compiles', () => {
		expect(typeof Schema).toBe('function');
	});

	it('parses', () => {
		schema = Schema.fromDom(sync(`<schema xmlns="http://purl.oclc.org/dsdl/schematron">
			<pattern>
				<rule context="//thunder">
					<let name="lightning" value="string(@foo)"/>
					<report test="$lightning = 'bar'"><value-of select="$lightning" /></report>
				</rule>
			</pattern>
		</schema>`));
		expect(schema).toMatchSnapshot();
	});

	it('validates', () => {
		expect(schema.validateDocument(sync(`<xml foo="err"><thunder foo="bar" /></xml>`))).toMatchSnapshot();
	});
});
