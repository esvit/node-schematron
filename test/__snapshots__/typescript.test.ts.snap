// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript parses 1`] = `
Schema {
  "defaultPhase": null,
  "patterns": Array [
    Pattern {
      "id": null,
      "rules": Array [
        Rule {
          "asserts": Array [
            Assert {
              "isReport": true,
              "message": Array [
                Object {
                  "$type": "value-of",
                  "select": "$lightning",
                },
              ],
              "test": "$lightning = 'bar'",
            },
          ],
          "context": "//thunder",
          "variables": Array [
            Variable {
              "name": "lightning",
              "value": "string(@foo)",
            },
          ],
        },
      ],
      "variables": Array [],
    },
  ],
  "phases": Array [],
  "title": null,
  "variables": Array [],
}
`;

exports[`typescript validates 1`] = `
Array [
  Result {
    "context": <thunder
      foo="bar"
    />,
    "isReport": true,
    "message": "bar",
  },
]
`;
